pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -241,8 +243,24 @@ func (ssr *Snapshotter) takeFullSnapshot() (*snapstore.Snapshot, error) {
 				Message: fmt.Sprintf("failed to create etcd snapshot: %v", err),
 			}
 		}
+
+		// compressionSuffix is useful in backward compatibility(restoring from non-compressed snapshots).
+		// it is also helpful in inferring which compression Policy to be used to decompress the snapshot.
+		compressionSuffix := compressor.GetCompressionSuffix(ssr.compressionConfig.Enabled, ssr.compressionConfig.CompressionPolicy)
 		ssr.logger.Infof("Successfully opened snapshot reader on etcd")
-		s := snapstore.NewSnapshot(snapstore.SnapshotKindFull, 0, lastRevision)
+		s := snapstore.NewSnapshot(snapstore.SnapshotKindFull, 0, lastRevision, compressionSuffix)
+
+		// if compression is enabled
+		//    then compress the snapshot.
+		if ssr.compressionConfig.Enabled {
+			ssr.logger.Info("starting the Compression of full snapshot")
+			startCompressTime := time.Now()
+			rc = compressor.CompressSnapshots(rc, ssr.compressionConfig.CompressionPolicy)
+			totalCompressTime := time.Now().Sub(startCompressTime).Seconds()
+			ssr.logger.Infof("Successfully compresses the full snapshot in %v seconds", totalCompressTime)
```

Comment:
This should be logged after the `ssr.store.Save()` call where the stream is read -> https://github.com/gardener/etcd-backup-restore/pull/293/files?file-filters%5B%5D=.go#diff-5c59e1a2063c6043cf7cad27aee199b833950c80ec75202ac69699216210a2f1R265

Reply:
1. I think there is already a separate timer ... which actually measures the time of upload. and with help of this we can measure both time... compression time as well as upload to object store time separately .
2. And When Compression is not enabled ... in that case your log still get printed but it should not.

Reply:
> I think there is already a separate timer which actually measures the time of upload and with help of this we can measure the time for both (compression time as well as upload to object store time) separately.

Yes. This is enough.

> Both timer will be helpful from future prospective like we can debug which one is taking more time either compression or upload to obj store.

No. The compression timer is not required because it won't measure only the time to construct a pipe and a compressing writer which will practically be instantaneous and not particularly meaningful.